web/src/components/__tests__/StatsFilters.test.tsx

```diff
@@ -0,0 +1,78 @@
+import { describe, it, expect, vi, beforeEach } from 'vitest';
+import { render, screen, fireEvent } from '@testing-library/react';
+
+// Mock useMediaQuery before importing StatsFilters
+vi.mock('react-responsive', () => ({
+  useMediaQuery: vi.fn(),
+}));
+
+interface WeekPickerProps {
+  mobile: boolean;
+  onChange?: (week: number) => void;
+  [key: string]: unknown;
+}
+
+vi.mock('../LeaguePicker', () => ({
+  __esModule: true,
+  default: ({ mobile, ...props }: { mobile: boolean; [key: string]: unknown }) => (
+    <div data-testid="league-picker" data-mobile={mobile} {...props}>
+      LeaguePicker
+    </div>
+  ),
+}));
+
+vi.mock('../WeekPicker', () => ({
+  __esModule: true,
+  default: ({ mobile, onChange, ...props }: WeekPickerProps) => (
+    <button
+      data-testid="week-picker"
+      data-mobile={mobile}
+      onClick={() => onChange?.(2)}
+      onKeyDown={e => {
+        if (e.key === 'Enter' || e.key === ' ') {
+          onChange?.(2);
+        }
+      }}
+      {...props}
+    >
+      WeekPicker
+    </button>
+  ),
+}));
+
+import StatsFilters from '../StatsFilters';
+import { useMediaQuery } from 'react-responsive';
+
+describe('StatsFilters', () => {
+  beforeEach(() => {
+    vi.clearAllMocks();
+  });
+
+  it('renders desktop filters', () => {
+    (useMediaQuery as unknown as ReturnType<typeof vi.fn>).mockReturnValue(false);
+    const changeWeek = vi.fn();
+    render(<StatsFilters data={{ week: 1, weeks: [1, 2, 3] }} changeWeek={changeWeek} />);
```

Comment:
not sure about these tests... the component doesn't do that much and the actual picker component tests are decent. This data is not realistic and it mocks out the pickers so the only thing being tested here is the responsive switch which really needs proper visual QA anyways.